File: two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let complement: { [key: number]: number } = {};  

    for (let i = 0; i < nums.length; i++) {
        if (complement[target - nums[i]] != undefined) {
            return [complement[target - nums[i]], i];
        }
        complement[nums[i]] = i;
    }
    return [];
};

File: two-sum/two-sum.java
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> complement = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (complement.get(target - nums[i]) != null) {
                return new int[]{complement.get(target - nums[i]), i};
            }
            complement.put(nums[i], i);
        } 
        return null;
    }
}